types/fixed.go

```diff
@@ -88,7 +89,11 @@ func (s *FixedDefinition) ResolveReferences(n *Namespace) error {
 
 func (s *FixedDefinition) Schema(names map[QualifiedName]interface{}) interface{} {
 	name := s.name.String()
-	names[s.name] = 1
+
+	// Add a small hash suffix to the name to avoid name collisions
```

Comment:
That works - could also pass name to hash ala `name = hash(name)` and have it append the `_` too